src/Automation/Automation/Common/AutomationPSClientSoftwareUpdateConfiguration.cs

```diff
@@ -109,7 +109,7 @@ public SoftwareUpdateConfiguration CreateSoftwareUpdateConfiguration(string reso
                                 : null,
                             ExcludedPackageNameMasks = updateConfig.Linux != null ? updateConfig.Linux.ExcludedPackageNameMasks : null,
                             IncludedPackageNameMasks = updateConfig.Linux != null ? updateConfig.Linux.IncludedPackageNameMasks : null,
-                            RebootSetting = updateConfig.Windows != null ? updateConfig.Windows.rebootSetting.ToString() : RebootSetting.IfRequired.ToString(),
+                            RebootSetting = updateConfig.Linux != null ? updateConfig.Linux.rebootSetting.ToString() : RebootSetting.IfRequired.ToString(),
                         },
```

Comment:
should we support the scenario that updateConfig.Windows is not null and updateConfig.Linux is null?

Reply:
@erich-wang  Can you please explain little bit more? As per my understanding, we need to check updateConfig.Windows not null in case of windows only. In case of linux, just check updateConfig.Linux not null.

Reply:
@assing, I just took a look at whole function and understand this code is only for Linux, LGTM now.